### ISSUE TITLE
Networking issue in only one copy of Ubuntu 20.04 WSL 2 Distro, Win 10 Firewall Public profile off is workaround

### ISSUE THREAD

**Issue by bc1121** [microsoft/WSL#6139]
Only for the past couple of weeks (not prior), when I run
     xdg-open https://microsoft.com
it takes several minutes for the website to appear in Chrome in Windows. Typically and in all other WSL distros on my machine, the website pops up in Windows in less than a second.

The delay only occurs in one apt updated/upgraded copy of Ubuntu 20.04 (1 of 3 Ubuntu WSL 2 distros) and does not occur in a another apt updated/upgraded Ubuntu 20.04 WSL distro that is freshly initialized with no other software on it on the same machine; or a semi fresh 18.04 for that matter.

Everything else seems to be running fine on the Ubuntu 20.04 WSL distro with the issue. This distro is my daily driver used for php development via ddev and docker.

The symptoms of the issue do not present when the Windows Defender Firewall Public profile is turned off. There is never an error and no packets are ever blocked based on win def firewall logs. The command always completes successfully but with a long delay on this one WSL distro when the win def public network firewall profile is turned on.

I had some success with a custom rule allowing inbound port 6000 in the Win Def Firewall public profile but that success was short lived. 

I suspect that the ARP tables are corrupted (or something in the neighborhood) and an Ubuntu network reset would correct the issue but I am unsure of the WSL recommended method to accomplish this without breaking Docker integration and I am also unsure if this was caused by a bug or just a normal possibility that can occur when using so much new tech together (WSL, Docker, Ubuntu, DDEV-Local, Windows Defender Firewall, xdg-utils, xdg-open)

I am curious if the WSL team would recommend troubleshooting this one from...
    #1 Win Def Firewall Public profile rules
       or
    #2 from within the WSL Ubuntu network stack.

I am sure wiping out WSL distro and starting over would fix it but would like to find out if there is a less drastic method of troubleshooting that takes less time than rebuilding the dev environment in the wsl distro.

# Environment

Windows build number: Microsoft Windows [Version 10.0.19041.572] 
Your Distribution version: Ubuntu 20.04
The issue is on WSL 2

# Steps to reproduce

sudo apt xdg-utils
xdg-open https://microsoft.com                        - or any website causes a long several minute delay on only one Ubuntu 20.04 WSL distro

The long delay can be relieved by turning off the Win Def Firewall Public Profile

---Wireshark Packet capture---
1	0.000000	172.19.188.7	172.19.176.1	TCP	74	54750 → 6000 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 SACK_PERM=1 TSval=524255763 TSecr=0 WS=128
2	1.019061	172.19.188.7	172.19.176.1	TCP	74	[TCP Retransmission] 54750 → 6000 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 SACK_PERM=1 TSval=524256782 TSecr=0 WS=128
3	3.098889	172.19.188.7	172.19.176.1	TCP	74	[TCP Retransmission] 54750 → 6000 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 SACK_PERM=1 TSval=524258862 TSecr=0 WS=128
4	5.018988	Microsof_6b:6c:bf	Microsof_38:28:31	ARP	42	Who has 172.19.176.1? Tell 172.19.188.7
5	5.019006	Microsof_38:28:31	Microsof_6b:6c:bf	ARP	42	172.19.176.1 is at 00:15:5d:38:28:31
6	7.178969	172.19.188.7	172.19.176.1	TCP	74	[TCP Retransmission] 54750 → 6000 [SYN] Seq=0 Win=64240 Len=0 MSS=1460 SACK_PERM=1 TSval=524262942 TSecr=0 WS=128
7	9.629218	fe80::c481:8553:a2ab:fcbd	ff02::1:ffc2:eeab	ICMPv6	86	Neighbor Solicitation for fe80::473:2653:52c2:eeab from 00:15:5d:38:28:31

**WSL logs**: 
https://gist.github.com/bc1121/2eae6c73e0321b7d7b45fafacea88d0c

#  Expected behavior

Website opens in Chrome on Windows in about 1 second

# Actual behavior

Long several minute delay ending in success with no errors or blocked packets. The delay is the issue.

**Comment by sirredbeard** [comment]
After running `xdg-open https://microsoft.com` I see a few dbus processes start up and then nothing happens at all for me.

The fact it's influenced by Windows Defender being enabled is strange.

For now I would recommend using `wslview https://microsoft.com` from wslutilities.

**Comment by sirredbeard** [comment]
> and then nothing happens at all for me.

Actually it eventually opens the link in Lynx, which I have installed and must be handling my web links.

If you don't have another browser installed then wslview will be registered as your web link handler, which is why it does work.

There seems to be some sort of hang up in dbus. It happens even when I have dbus launched and running.

**Comment by bc1121** [comment]
I have discovered the same issue on another computer also running Windows 10 WSL 2 Ubuntu 20.04. Also my main WSL 2 machine on that computer. It must have quit working around the same time as the laptop because it was running fine right before the laptop started acting up and it wasn't working right today (same long delay but no error) which is the first time I've tried it since the laptop started acting up.

It caused my to wonder if it was a networking issue that had to do with Windows or Ubuntu updates. The curious thing is that I can spin up a new Ubuntu WSL 2 machine and it works fine. I just wonder if maybe something got updated while the machines were running and it caused some strange networking cache errors because the new machines and the machines that were not running were not affected.

wslview looks interesting. I'll check that out!
